mergin/client.py

```diff
@@ -1061,3 +1062,34 @@ def has_writing_permissions(self, project_path):
         """
         info = self.project_info(project_path)
         return info["permissions"]["upload"]
+
+    def reset_local_changes(self, directory: str, files_to_reset: typing.List[str] = None) -> None:
+        """
+        Reset local changes to either all files or only listed files.
+        Added files are removed, removed files are brought back and updates are discarded.
+
+        :param directory: Project's directory
+        :type directory: String
+        :param files_to_reset List of files to reset, relative paths of file
+        :type files_to_reset: List of strings, default None
+        """
+        all_files = files_to_reset is None
+
+        mp = MerginProject(directory)
+
+        push_changes = mp.get_push_changes()
+
+        # remove all added files
+        for file in push_changes["added"]:
+            if all_files or file["path"] in files_to_reset:
+                os.remove(mp.fpath(file["path"]))
+
+        # update files get override with previous version
+        for file in push_changes["updated"]:
+            if all_files or file["path"] in files_to_reset:
+                mp.geodiff.make_copy_sqlite(mp.fpath_meta(file["path"]), mp.fpath(file["path"]))
+
+        # removed files are redownloaded
+        for file in push_changes["removed"]:
+            if all_files or file["path"] in files_to_reset:
+                self.download_file(directory, file["path"], mp.fpath(file["path"]))
```

Comment:
I am wondering if it is not worth adding also MerginClient.download_files() variant - in case there are multiple updated/removed files, we would be currently doing a lot of duplicate work (querying project info) and doing the downloads sequentially when they could be done parallel...

Reply:
This would need also new function like `download_file_async()` right? For downloading multiple files at once...Or I am missing something?